backend/module/eCampApi/src/eCampApi/V1/Factory/Config.php

```diff
@@ -0,0 +1,264 @@
+<?php
+
+namespace eCampApi\V1\Factory;
+
+class Config {
+    private $nameSingular;
+    private $namePlural;
+    private $routeName;
+    private $routeIdentifierName;
+    private $endpoint;
+    private $entityHttpMethods;
+    private $collectionHttpMethods;
+    private $collectionQueryWhitelist = ['page_size', 'page'];
+    private $inputFilterItems = [];
+
+    public function __construct(string $nameSingular) {
+        $this->nameSingular = $nameSingular;
+    }
+
+    public static function Create(string $nameSingular, ?string $namePlural = null): Config {
```

Comment:
The start with a capital letter for a method comes from c#.
It's not php like, even if its a static factory method.

Reply:
#1018